fennel/client/client.py

```diff
@@ -779,3 +776,15 @@ def _s3_connector_dict(s3: S3Connector) -> Dict[str, Any]:
         "db": {"S3": {"creds": creds_json}},
     }
     return s3_table
+
+
+def _validate_branch_name(branch: str):
+    """
+    Branch name should only contain alphanumeric characters, hyphens or underscores.
```

Comment:
Why are we adding this restriction? I think it's important to support "/" since many branches will be of the form "author/something". I think we should stick with the equivalent of git branch names: https://stackoverflow.com/questions/3651860/which-characters-are-illegal-within-a-branch-name

Reply:
Yeah that is precisely why I have added this restriction. author/name wont work , since it can break at a bunch of places, like we pass it in the URLs etc

Reply:
Currently the directive I planning is - <author_name>-<branch_name>